Jenkinsfile

```diff
@@ -63,22 +64,86 @@ pipeline {
         }
       }
     }
-    stage ('Run application tests') {
+    stage ('Build and Test') {
       when {
         expression { env.BRANCH_NAME != 'latest' }
       }
-      agent {
-        docker {
-          image "${nodeImage}"
-          label nodeName
-          args '-u root -v /etc/pki:/certs'
+      parallel {
```

Comment:
The stage `stage('Test Development')` here as well as that below for the `latest` branch are identical and should remain that way. This should be in a reusable function at the top of this file. This will prevent us from having regressions - where one instance is updated and another is forgotten about.

Likewise the parts of the `stage('Test Production')` that can be re-used, should be in a shared function.

Reply:
Yes great idea. However I would like to first fix the race condition bug and then do a proper cleanup and refactor step. I have added the relevant issue to simorgh: https://github.com/bbc/simorgh/issues/1626

Reply:
Thanks for writing that up.  Happy to have the refactor in the separate issue, to prioritise fixing this issue in our pipeline.